packages/common/driver-definitions/src/driverError.ts

```diff
@@ -3,11 +3,103 @@
  * Licensed under the MIT License.
```

Comment:
Question for reviewers: Currently `IDriverErrorBase`'s `errorType` property is typed as `DriverErrorType`. I can't change this in this PR, as it would be breaking. But for the eventual follow-up on `next`, should this be generalized as `string` instead? That is how the Container error base is typed.

Reply:
I think it should be limited to DriverErrorType only as the error is of type IDriverErrorBase only. These represents error coming from driver only.

Reply:
That means that changes to the underlying enum (string union) are breaking in terms of the error interface, though. `string` would be more flexible to changes over time. Individual error classes still specify explicit string values, so that behavior doesn't change. I guess it's not clear to me what the value of the constraint is.

Reply:
wouldn't that make it equivalent to IAnyDriverError?

Reply:
what about something like this:
```
export interface IAnyDriverError extends Omit<IDriverErrorBase<DriverErrorType>, "errorType"> {
	readonly errorType: string;
}

/**
 * Base interface for all errors and warnings
 */
export interface IDriverErrorBase<T extends DriverErrorType> {
	/**
	 * Classification of what type of error this is, used programmatically by consumers to interpret the error
	 */
	readonly errorType: T;
```
```
export interface IThrottlingWarning extends IDriverErrorBase<DriverErrorType.throttlingError> {
	readonly errorType: DriverErrorType.throttlingError;
	readonly retryAfterSeconds: number;
}
```

Reply:
this would help keep additions isolated to just IDriverErrorBase

Reply:
Is the idea then that external consumers would observe `IAnyDriverError`, and internally we would implement `IDriverErrorBase` for specific error classes? I like that. We could mark `IDriverErrorBase` and our implementations as `@internal`.

Reply:
i don't know where we use IAnyDriverError and IDriverErrorBase currently. ,my suggestion isn't the change usages, internal or external, its about supporting inheritance from IDriverErrorBase in a way that keeps inheritors isolated from unrelated modification to the DriverErrorTypes

Reply:
i'm not even really sure the generic is necessary poking at it, the existing code seems to overwrite the base property. enum might still cause a problem, but const object might just solve that without other changes

Reply:
probably need some prototyping to see how the types play out

Reply:
Oh, okay. I missed that `IAnyDriverError` was something that already existed - thought that was a new suggestion. Makes sense. Probably worth updating the container error hierarchy to match?

Reply:
I'll take that as the plan of record then. Resolving this for now, though folks are welcome to re-open with differing opinions.